src/FondOfSpryker/Zed/ThirtyFiveUp/Business/Model/Mapper/ThirtyFiveUpOrderMapper.php

```diff
@@ -0,0 +1,217 @@
+        $thirtyFiveUpOrderTransfer
+            ->setOrderReference($saveOrderTransfer->getOrderReference())
+            ->setSalesOrderId($saveOrderTransfer->getIdSalesOrder());
+
+        return $thirtyFiveUpOrderTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\ThirtyFiveUpOrderItemTransfer[]
+     */
+    protected function resolveThirtyFiveUpItems(QuoteTransfer $quoteTransfer): array
+    {
+        $thirtyFiveUpItems = [];
+        $knownVendor = $this->config->getKnownVendor();
+        $defaultLocale = $this->localeFacade->getCurrentLocaleName();
+        foreach ($this->groupItems($quoteTransfer->getItems()) as $itemTransfer) {
+            $attributes = $itemTransfer->getAbstractAttributes();
+
+            if (array_key_exists($defaultLocale, $attributes)) {
+                $thirtyFiveUpItems = $this->getItemFromAttributes(
+                    $attributes[$defaultLocale],
+                    $knownVendor,
+                    $thirtyFiveUpItems,
+                    $itemTransfer
+                );
+            }
+            if (array_key_exists(static::FALLBACK_LOCALE, $attributes)) {
+                $thirtyFiveUpItems = $this->getItemFromAttributes(
+                    $attributes[static::FALLBACK_LOCALE],
+                    $knownVendor,
+                    $thirtyFiveUpItems,
+                    $itemTransfer
+                );
+            }
+        }
+
+        return $thirtyFiveUpItems;
+    }
+
+    /**
+     * @param \ArrayObject|\Generated\Shared\Transfer\ItemTransfer[] $items
+     *
+     * @return \Generated\Shared\Transfer\ItemTransfer[]
+     */
+    protected function groupItems(ArrayObject $items): array
+    {
+        /** @var \Generated\Shared\Transfer\ItemTransfer[] $groupedItems */
+        $groupedItems = [];
+        foreach ($items as $itemTransfer) {
+            $sku = $itemTransfer->getSku();
+            if (array_key_exists($sku, $groupedItems)) {
+                $groupedItems[$sku]->setQuantity($itemTransfer->getQuantity() + $groupedItems[$sku]->getQuantity());
+
+                continue;
+            }
+
+            $groupedItems[$sku] = clone $itemTransfer;
+        }
+
+        return $groupedItems;
+    }
+
+    /**
+     * @param string $vendorName
+     *
+     * @return \Generated\Shared\Transfer\ThirtyFiveUpVendorTransfer
+     */
+    protected function mapVendor(string $vendorName): ThirtyFiveUpVendorTransfer
+    {
+        return (new ThirtyFiveUpVendorTransfer())
+            ->setName($vendorName);
+    }
+
+    /**
+     * @param string $haystack
+     * @param string $needle
+     *
+     * @return bool
+     */
+    protected function stringEndsWith(string $haystack, string $needle): bool
+    {
+        $length = strlen($needle);
+        if ($length === 0) {
+            return true;
+        }
+
+        return (substr($haystack, -$length) === $needle);
+    }
```

Comment:
Good that this will be dropped with PHP8 :D might be worth that we create a helper package for str functions until we can switch